### ISSUE TITLE
Craft 4 Support?

### ISSUE THREAD

**Issue by DynamiteGoesBoom** [FosterCommerce/snipcart-craft-plugin#48]
I'm just curious if y'all had plans for a Craft 4 port of this plugin? I'm sure you do but was curious if you had an ETA on that at all.

Thanks again!

**Comment by DynamiteGoesBoom** [comment]
Can we get an update on this?

**Comment by sjcallender** [comment]
It's in active development. We're fixing some legacy issues along the way. This month is the goal.

**Comment by sjcallender** [comment]
We're still working on it. It's a chore. We're in communication with the Craft team about some best practices for migrations we've had to make. One issue reveals a new one. We want to get it right.

I wish I could give a timeline, but that'd be more hopeful than realistic.

**Comment by DynamiteGoesBoom** [comment]
Thanks for the update @sjcallender and all the work your team is doing! What if there are some projects that don't require any migrations. Do you have a solid branch for Craft 4 if there aren't any migrations that need to run?

Thanks again!

**Comment by sjcallender** [comment]
Migrations are needed for all current Craft 3 builds because we have to deprecate some work moving into the Craft 4 version. There's no stable Craft 4 version because we haven't yet solved what the final state for the needed migrations will be.

**Comment by DynamiteGoesBoom** [comment]
Makes sense and thanks again for the update!

**Comment by wbdzynr** [comment]
Hey all - just checking to see if there is any update on this?

**Comment by javangriff** [comment]
@sjcallender any update on the progress of updating this for Craft 4?

**Comment by fantasticmachine** [comment]
Hey all, apologies for the slow response here.

We have a working version for Craft 4. This has involved a change to how the data is stored to better fit with Craft itself. We are currently (still) figuring out how to migrate old data into the new format to make it possible to upgrade without losing your current product set up.

**Comment by tobystokes** [comment]
Out of interest, if I was _not_ using the this plugin's fieldtype to store product data, is the develop-craft4 branch worth a test?

**Comment by fantasticmachine** [comment]
@tobystokes Yes you should be fine to use it in that case.

In fact you could still use the field type to store product data, you'd just need to re-enter anything that you had set up in previous versions.

The difficulty at the moment is converting the old way to the new way.

**Comment by sjcallender** [comment]
I don't want to jinx things, but we are very close to _finally_ having a proper Craft 4-compatible version released. We've crossed the main migration hurdle and are now just triple-checking everything again and updating tests.

**Comment by brettburwell** [comment]
Great news. Thanks for the update @sjcallender!

**Comment by tobystokes** [comment]
Good news indeed.
BTW, we are actually running a live Craft 4 site with a [fork of this plugin](https://github.com/FosterCommerce/snipcart-craft-plugin/compare/develop-craft4...simplygoodwork:snipcart-craft-plugin:develop-craft4) – we had to make a few minor amends, which we were planning to PR to you properly, but may have been fixed in the meantime anyway... do feel free to merge them in, I think they are hopefully self-explanatory.

**Comment by johnnynotsolucky** [comment]
@tobystokes great! I will include anything that's still needed in our migration.

**Comment by sjcallender** [comment]
Hear ye! Hear ye! The official Craft 4 version of this plugin is officially ready for you to test. 

We'll plan to officially release it next week pending any issues you all discover and report.

To try it out, run the following command. And thanks for your incredible patience.

```
composer require "fostercommerce/craft-snipcart:dev-craft4 as 2.0.0"
```

See the [changelog](https://github.com/FosterCommerce/snipcart-craft-plugin/blob/craft4/changelog.md#20----2023-09-11).

**Comment by sjcallender** [comment]
Officially released v2.0.0 with https://github.com/FosterCommerce/snipcart-craft-plugin/pull/54
